src/Jenssegers/Mongodb/Eloquent/Builder.php

```diff
@@ -167,9 +168,10 @@ protected function addHasWhere(EloquentBuilder $hasQuery, Relation $relation, $o
         $query = $hasQuery->getQuery();
 
         // Get the number of related objects for each possible parent.
+        $relations = $query->pluck($relation->getHasCompareKey());
         $relationCount = array_count_values(array_map(function ($id) {
             return (string) $id; // Convert Back ObjectIds to Strings
-        }, $query->pluck($relation->getHasCompareKey())));
+        }, is_array($relations) ? $relations : $relations->toArray()));
```

Comment:
This tests fail without this check:
- RelationsTest.php:384
- RelationsTest.php:433

Error: `array_map(): Argument #2 should be an array`

Laravel `has` returns arrays instead of Collections!